enter-chroot.sh

```diff
@@ -9,4 +9,17 @@ then
     exit 1
 fi
 
-sudo systemd-nspawn -M homeworld --bind $(pwd):/homeworld:norbind -u "$USER" -a -D "$HOMEWORLD_CHROOT" bash -c "cd /h/ && exec bash"
+ORIG_PWD="$(pwd)"
+cd "$(dirname "$0")"
+ORIG_REL="$(realpath --relative-to "$(pwd)/building" "${ORIG_PWD}")"
+if [[ "${ORIG_REL}" =~ \.\. ]]
+then
+	ORIG_REL=""
+fi
+if [ -e "$HOME/.gnupg/pubring.kbx" ]
+then
+	cp "$HOME/.gnupg/pubring.kbx" "$HOMEWORLD_CHROOT/home/$USER/.gnupg/pubring.kbx"
+	cp "$HOME/.gnupg/trustdb.gpg" "$HOMEWORLD_CHROOT/home/$USER/.gnupg/trustdb.gpg"
+	cp -R "$HOME/.gnupg/private-keys-v1.d/"* "$HOMEWORLD_CHROOT/home/$USER/.gnupg/private-keys-v1.d"
```

Comment:
I'm concerned about copying the user's private keys into the build chroot.  Why is this necessary?

Reply:
It's necessary in order to sign packages within the chroot, which is currently how the upload process works.

Reply:
I think this could be done in a better way.
- I don't think `private-keys-v1.d` is a good interface to use; we should use `gpg --export-secret-key` instead.
- Along the same lines, we should move only the key we need for signing.

Reply:
We don't know what key we'll need for signing at the point where we enter the chroot.

Reply:
How does the upload process determine which key to use?

Reply:
It's based on the branch specified by the user, which is usually only specified once they're already within the chroot.

Reply:
Can we at least take only the ones listed in `(global-)?signing-keys`?

Reply:
We could try.

In any case, I tried this, and it turns out that then the user has to manually decrypt the relevant pgp keys to insert them into the chroot, and that's not especially fun. It'll require a lot of complicated synchronization code to do this, and since you already LGTM'd this part of the code in the other PR, I'm separating this out into a separate bug, #287, to be acted on later.